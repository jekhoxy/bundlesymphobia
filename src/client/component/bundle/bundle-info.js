import React from "react";
import { BsInfoCircle } from "react-icons/bs";
import { FaGithub, FaNpm } from "react-icons/fa";
import { GoPackage } from "react-icons/go";
import styled from "styled-components";
import { Icon } from "../layout/icon";

// STYLED
const Container = styled.div`
  font-size: small;
  color: gray;
  background-color: #80808021;
  display: flex;
  align-items: center;
  justify-content: space-around;
`;

const SubContainer = styled.div`
  display: flex;
  align-items: center;

  margin: 0 5px;

  & * {
    margin: 0 2px;
  }
`;

const NPMLink = styled(Icon)`
  font-size: 30px;

  &:hover {
    color: red;
  }
`;

const GitHubLink = styled(Icon)`
  font-size: 20px;

  &:hover {
    color: black;
  }
`;

// COMPONENT
/**
 * Component that displays Bundle info such as name, description, dependencies and repository
 *
 */
export default function BundleInfo({
  name,
  description,
  dependencies,
  repository,
}) {
  const { url } = repository;
  const urlRepo = url ? url.slice(4) : "";

  return (
    <Container>
      <SubContainer>
        <Icon>
          <BsInfoCircle />
        </Icon>
        <i>{description} </i>
      </SubContainer>
      <SubContainer>|</SubContainer>
      <SubContainer>
        <Icon>
          <GoPackage />
        </Icon>
        {dependencies} dependencies{" "}
      </SubContainer>
      <SubContainer>|</SubContainer>
      <SubContainer>
        <a
          href={`https://npmjs.org/package/${name}`}
          target="_blank"
          rel="noopener noreferrer">
          <NPMLink>
            <FaNpm />
          </NPMLink>
        </a>
        <a href={urlRepo} target="_blank" rel="noopener noreferrer">
          <GitHubLink>
            <FaGithub />
          </GitHubLink>
        </a>
      </SubContainer>
    </Container>
  );
}
