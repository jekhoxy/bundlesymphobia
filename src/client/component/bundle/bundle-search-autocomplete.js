import PropTypes from "prop-types";
import React, { useEffect, useState } from "react";
import { BsSearch } from "react-icons/bs";
import { FiRefreshCw } from "react-icons/fi";
import { connect } from "react-redux";
import { useHistory } from "react-router-dom";
import styled from "styled-components";
import {
  BundleStatus,
  clearBundleAction,
  getBundleSuggestionsAction,
} from "../../redux/action/bundle-action";
import Error from "../layout/error";
import { RotateIcon } from "../layout/icon";

// STYLES
const Container = styled.div`
  width: 650px;
  border: 1px solid #9e9e9e80;
  border-radius: 10px;
`;

const Form = styled.form`
  width: 90%;
  padding: 10px 25px;
  display: flex;
  align-items: center;
`;

const SearchInput = styled.input`
  border: none;
  width: 100%;
  font-size: xx-large;
  caret-color: rgb(65, 171, 93);
  font-family: inherit;

  &:focus {
    outline: none;
  }
`;

const SuggestionsContainer = styled.div`
  display: ${(props) => (props.open ? "block" : "none")};
  max-height: 150px;
  overflow: auto;
  background-color: white;
  border: 1px solid #9e9e9e80;
`;

const Suggestion = styled.div`
  background-color: ${(props) => (props.selected ? "#98e9ff" : "transparent")};
`;

const SuggestionName = styled.div`
  font-weight: 600;
  padding: 5px 0px 2px 25px;
  font-size: 12px;
`;

const SuggestionDesc = styled.div`
  font-size: 10px;
  padding: 2px 0px 5px 25px;
  color: gray;
`;

const DivError = styled.div`
  text-align: center;
`;

let lastTimeout;

// COMPONENT
/**
 *
 * Component to seek a bundle. Display an input and autocomplete list with matched bundles.
 * When choosing a bundle, send to result page.
 */
export function BundleSearchAutocomplete(props) {
  const { suggestions, bundle, version, fetching, error } = props;

  const history = useHistory();
  const [inputValue, setInputValue] = useState(
    bundle && version ? `${bundle}@${version}` : ""
  );
  const [open, setOpen] = useState(false);
  const [selection, setSelection] = useState(null);

  // Update on suggestions props
  useEffect(() => {
    setOpen(suggestions.length > 0);
    setSelection(suggestions.length > 0 ? suggestions[0] : null);
  }, [suggestions]);

  // Update on bundle/version input
  useEffect(() => {
    if (bundle && version) {
      setInputValue(bundle + "@" + version);
    }
  }, [bundle, version]);

  // Delay getting suggestions after update input value
  useEffect(() => {
    // Delay getting suggestions
    if (lastTimeout) {
      clearTimeout(lastTimeout);
    }

    lastTimeout = setTimeout(() => {
      if (inputValue.length >= 2) {
        props.getBundleSuggestionsAction(inputValue);
      }
    }, 250);
  }, [inputValue]);

  // CALLBACKS
  function onChange(event) {
    const { value } = event.target;

    setInputValue(value);
  }

  function onSelectBundle(name, version) {
    history.push(`/results?bundle=${name}&version=${version}`);
    props.clearBundleAction();
  }

  function onSubmit() {
    if (selection) {
      onSelectBundle(selection.name);
    }
  }

  // RENDER
  return (
    <Container>
      <Form onSubmit={onSubmit}>
        <SearchInput
          placeholder="Find a package"
          type="text"
          value={inputValue}
          onChange={onChange}
          onFocus={() => setOpen(true)}
          onBlur={(e) => setOpen(false)}></SearchInput>

        <RotateIcon rotate={fetching}>
          {fetching ? <FiRefreshCw /> : <BsSearch />}
        </RotateIcon>
      </Form>

      <SuggestionsContainer open={open}>
        {suggestions.map((suggestion) => (
          <Suggestion
            selected={selection && selection.name === suggestion.name}
            key={suggestion.name}
            onMouseDown={() =>
              onSelectBundle(suggestion.name, suggestion.version)
            }
            onMouseEnter={() => setSelection(suggestion)}>
            <SuggestionName>{suggestion.name} </SuggestionName>

            <SuggestionDesc>
              <i>{suggestion.description}</i>
            </SuggestionDesc>
          </Suggestion>
        ))}
      </SuggestionsContainer>

      {error ? (
        <DivError>
          <Error message={error}></Error>
        </DivError>
      ) : null}

      {props.children}
    </Container>
  );
}

BundleSearchAutocomplete.propTypes = {
  name: PropTypes.string,
  version: PropTypes.string,
  fetching: PropTypes.bool,
};

BundleSearchAutocomplete.defaultProps = {
  name: "",
  version: "",
  fetching: false,
};

const mapStateToProps = (state, ownProps) => {
  return {
    suggestions: state.suggestions,
    suggestionStatus: state.suggestionStatus,
    error: state.error,
    fetching: state.suggestionStatus === BundleStatus.inProgress,
    ...ownProps,
  };
};

const mapDispatchToProps = {
  getBundleSuggestionsAction,
  clearBundleAction,
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(BundleSearchAutocomplete);
