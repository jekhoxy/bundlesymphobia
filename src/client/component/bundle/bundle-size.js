import React, { useEffect, useState } from "react";
import styled from "styled-components";
import { formatValue } from "../../util/sizeUtil";

// STYLED
const Container = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
`;

const ValueContainer = styled.div`
  display: flex;
  align-items: flex-end;

  & > label {
    margin: 0 2px;
  }
`;

const Title = styled.h4`
  margin-top: -5px;
`;

const Value = styled.label`
  font-size: xxx-large;
`;

const Unit = styled.label`
  color: gray;
  font-size: xx-large;
  font-weight: 700;
`;

/**
 *
 * Display a HDD size with unit (default kB).
 * The size is incremented from 0 to size (like a counter)
 *
 */
export default function BundleSize({ title, size, unit = "kB" }) {
  const step = size / 10;
  const [value, setValue] = useState(0);

  useEffect(() => {
    setValue(0);
  }, [size]);

  if (value < size) {
    setTimeout(() => setValue(Math.min(value + step, size)), 50);
  }

  return (
    <Container>
      <ValueContainer>
        <Value>{formatValue(value)}</Value> <Unit>{unit}</Unit>
      </ValueContainer>
      <Title>{title}</Title>
    </Container>
  );
}
