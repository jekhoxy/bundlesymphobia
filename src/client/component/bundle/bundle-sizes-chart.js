import { ResponsiveBar } from "@nivo/bar";
import React from "react";
import { formatValue } from "../../util/sizeUtil";

/**
 * Display bundle sizes in a chart (bar chart).
 */
export default function BundleSizesChart({ sizes }) {
  const dataset = formatData(sizes);

  return (
    <ResponsiveBar
      data={dataset}
      keys={["initial", "minify", "gzip"]}
      indexBy="version"
      colors={["rgb(65, 182, 196)", "rgb(127, 205, 187)", "rgb(199, 233, 180)"]}
      labelTextColor={{ from: "color", modifiers: [["darker", 1.6]] }}
      label={(d) => formatValue(d.value) + "kB"}
      isInteractive={false}
      margin={{ bottom: 100 }}
      enableGridX={false}
      enableGridY={false}
      axisBottom={{
        tickSize: 5,
        tickPadding: 5,
        tickRotation: 90,
        legendPosition: "middle",
      }}
      axisLeft={null}
      legends={[
        {
          dataFrom: "keys",
          anchor: "bottom",
          direction: "row",
          itemWidth: 100,
          itemHeight: 20,
          translateX: 25,
          translateY: 80,
        },
      ]}
      animate={true}
    />
  );
}

function formatData(sizes) {
  return sizes.map((size) => {
    return {
      version: size.version,
      initial: size.initialSize,
      minify: size.minifySize,
      gzip: size.gzipSize,
    };
  });
}
