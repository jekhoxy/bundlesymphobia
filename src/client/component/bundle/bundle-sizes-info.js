import React from "react";
import styled from "styled-components";
import BundleSize from "./bundle-size";

// STYLED
const Container = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 100%;
`;

const SizesContainer = styled.div`
  display: flex;
  justify-content: space-around;
  width: 100%;
  margin: 20px 0px;

  & > div {
    margin: 0 25px;
  }
`;

/**
 * Display bundle info size in textual way
 */
export default function BundleSizesInfo({ size }) {
  return (
    <Container>
      <h2>bundle size</h2>
      <SizesContainer>
        <BundleSize title={"initial"} size={size.initialSize} />
        <BundleSize title={"minified"} size={size.minifySize} />
        <BundleSize title={"gzipped"} size={size.gzipSize} />
      </SizesContainer>
    </Container>
  );
}
