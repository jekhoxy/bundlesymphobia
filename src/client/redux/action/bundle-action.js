import { getBundleDetails, searchBundles } from "../../api/bundle-api";

/**
 * Status result for request
 */
export const BundleStatus = {
  unknown: "unknown",
  ok: "ok",
  error: "error",
  inProgress: "inProgress",
};

/**
 * Redux action types
 */
export const BundleActionTypes = {
  GET_BUNDLE_SUGGESTIONS: "GET_BUNDLE_SUGGESTIONS",
  GET_BUNDLE_DETAILS: "GET_BUNDLE_DETAILS",
  CLEAR_BUNDLE: "CLEAR_BUNDLE",
};

/**
 * Redux action to get bundle suggestions
 */
export const getBundleSuggestionsAction = (value) => {
  return (dispatch) => {
    dispatch({
      type: BundleActionTypes.GET_BUNDLE_SUGGESTIONS,
      status: BundleStatus.inProgress,
    });

    return searchBundles(value)
      .then((bundles) => {
        dispatch({
          type: BundleActionTypes.GET_BUNDLE_SUGGESTIONS,
          status: BundleStatus.ok,
          suggestions: bundles,
        });
      })

      .catch((error) =>
        dispatch({
          type: BundleActionTypes.GET_BUNDLE_SUGGESTIONS,
          status: BundleStatus.error,
          error: error.message,
        })
      );
  };
};

/**
 * Redux action to clear bundle suggestions and details
 */
export const clearBundleAction = () => {
  return {
    type: BundleActionTypes.CLEAR_BUNDLE,
  };
};

/**
 * Redux action to get bundle details
 */
export const getBundleDetailsAction = (bundleName) => {
  return (dispatch) => {
    dispatch({
      type: BundleActionTypes.GET_BUNDLE_DETAILS,
      status: BundleStatus.inProgress,
    });

    return getBundleDetails(bundleName)
      .then((details) => {
        return dispatch({
          type: BundleActionTypes.GET_BUNDLE_DETAILS,
          status: BundleStatus.ok,
          bundleDetails: details,
        });
      })

      .catch((error) =>
        dispatch({
          type: BundleActionTypes.GET_BUNDLE_DETAILS,
          status: BundleStatus.error,
          error: error.message,
        })
      );
  };
};
