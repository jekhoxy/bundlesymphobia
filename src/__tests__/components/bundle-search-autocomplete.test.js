import { fireEvent, render } from "@testing-library/react";
import React from "react";
import { BundleSearchAutocomplete } from "../../client/component/bundle/bundle-search-autocomplete";
import { BundleStatus } from "../../client/redux/action/bundle-action";

describe("Test BundleSearchAutocomplete component", () => {
  it("should render", () => {
    const component = render(<BundleSearchAutocomplete suggestions={[]} />);

    expect(component).toMatchSnapshot();
  });

  it("should open suggestions panel", async () => {
    const suggestions = [
      { name: "suggest1", description: "description 1" },
      { name: "suggest2", description: "description 2" },
      { name: "suggest3", description: "description 3" },
    ];

    const { getByPlaceholderText, queryAllByText } = render(
      <BundleSearchAutocomplete suggestions={suggestions} />
    );

    fireEvent.click(getByPlaceholderText("Find a package"));

    const suggestionsElements = queryAllByText(/suggest/i);
    expect(suggestionsElements.length).toBe(3);
  });

  it("should display the bundle name & version", () => {
    const { getByPlaceholderText } = render(
      <BundleSearchAutocomplete
        suggestions={[]}
        bundle="react"
        version="1.0.0"
      />
    );

    expect(getByPlaceholderText("Find a package").getAttribute("value")).toBe(
      "react@1.0.0"
    );
  });

  it("should display an error", () => {
    const { queryByText } = render(
      <BundleSearchAutocomplete
        suggestions={[]}
        suggestionStatus={BundleStatus.error}
        error="Error message"
      />
    );

    expect(queryByText("Error message")).not.toBeNull();
  });

  it("should display a refresh icon", () => {
    const component = render(
      <BundleSearchAutocomplete
        suggestions={[]}
        suggestionStatus={BundleStatus.inProgress}
      />
    );

    expect(component).toMatchSnapshot();
  });
});
