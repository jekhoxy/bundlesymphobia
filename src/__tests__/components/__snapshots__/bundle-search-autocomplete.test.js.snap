// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test BundleSearchAutocomplete component should display a refresh icon 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-AxhCb hpcrhp"
      >
        <form
          class="sc-AxhUy lexsms"
        >
          <input
            class="sc-AxgMl kkFbcO"
            placeholder="Find a package"
            type="text"
            value=""
          />
          <div
            class="sc-AxirZ sc-AxiKw kPBqlq"
            font-size="inherit"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M10.442 10.442a1 1 0 011.415 0l3.85 3.85a1 1 0 01-1.414 1.415l-3.85-3.85a1 1 0 010-1.415z"
                fill-rule="evenodd"
              />
              <path
                clip-rule="evenodd"
                d="M6.5 12a5.5 5.5 0 100-11 5.5 5.5 0 000 11zM13 6.5a6.5 6.5 0 11-13 0 6.5 6.5 0 0113 0z"
                fill-rule="evenodd"
              />
            </svg>
          </div>
        </form>
        <div
          class="sc-AxheI deggKy"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-AxhCb hpcrhp"
    >
      <form
        class="sc-AxhUy lexsms"
      >
        <input
          class="sc-AxgMl kkFbcO"
          placeholder="Find a package"
          type="text"
          value=""
        />
        <div
          class="sc-AxirZ sc-AxiKw kPBqlq"
          font-size="inherit"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M10.442 10.442a1 1 0 011.415 0l3.85 3.85a1 1 0 01-1.414 1.415l-3.85-3.85a1 1 0 010-1.415z"
              fill-rule="evenodd"
            />
            <path
              clip-rule="evenodd"
              d="M6.5 12a5.5 5.5 0 100-11 5.5 5.5 0 000 11zM13 6.5a6.5 6.5 0 11-13 0 6.5 6.5 0 0113 0z"
              fill-rule="evenodd"
            />
          </svg>
        </div>
      </form>
      <div
        class="sc-AxheI deggKy"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Test BundleSearchAutocomplete component should render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-AxhCb hpcrhp"
      >
        <form
          class="sc-AxhUy lexsms"
        >
          <input
            class="sc-AxgMl kkFbcO"
            placeholder="Find a package"
            type="text"
            value=""
          />
          <div
            class="sc-AxirZ sc-AxiKw kPBqlq"
            font-size="inherit"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M10.442 10.442a1 1 0 011.415 0l3.85 3.85a1 1 0 01-1.414 1.415l-3.85-3.85a1 1 0 010-1.415z"
                fill-rule="evenodd"
              />
              <path
                clip-rule="evenodd"
                d="M6.5 12a5.5 5.5 0 100-11 5.5 5.5 0 000 11zM13 6.5a6.5 6.5 0 11-13 0 6.5 6.5 0 0113 0z"
                fill-rule="evenodd"
              />
            </svg>
          </div>
        </form>
        <div
          class="sc-AxheI deggKy"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-AxhCb hpcrhp"
    >
      <form
        class="sc-AxhUy lexsms"
      >
        <input
          class="sc-AxgMl kkFbcO"
          placeholder="Find a package"
          type="text"
          value=""
        />
        <div
          class="sc-AxirZ sc-AxiKw kPBqlq"
          font-size="inherit"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M10.442 10.442a1 1 0 011.415 0l3.85 3.85a1 1 0 01-1.414 1.415l-3.85-3.85a1 1 0 010-1.415z"
              fill-rule="evenodd"
            />
            <path
              clip-rule="evenodd"
              d="M6.5 12a5.5 5.5 0 100-11 5.5 5.5 0 000 11zM13 6.5a6.5 6.5 0 11-13 0 6.5 6.5 0 0113 0z"
              fill-rule="evenodd"
            />
          </svg>
        </div>
      </form>
      <div
        class="sc-AxheI deggKy"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
